turbopack/crates/turbo-tasks-backend/src/utils/dash_map_multi.rs

```diff
@@ -144,28 +170,28 @@ where
                 }
             }
         };
-        let e1 = guard1
-            .raw_entry_mut()
-            .from_key(&key1)
-            .or_insert_with(|| (key1, SharedValue::new(insert_with())));
-        let key1 = e1.0 as *const K;
-        let value1 = e1.1.get_mut() as *mut V;
-        let e2 = guard2
-            .raw_entry_mut()
-            .from_key(&key2)
-            .or_insert_with(|| (key2, SharedValue::new(insert_with())));
-        let key2 = e2.0 as *const K;
-        let value2 = e2.1.get_mut() as *mut V;
+
+        let bucket1 = guard1
+            .find_or_find_insert_slot(h1, eq1, hash_entry)
+            .unwrap_or_else(|slot| unsafe {
+                // SAFETY: See first insert_in_slot call
+                guard1.insert_in_slot(h1, slot, (key1.clone(), SharedValue::new(insert_with())))
+            });
+        let bucket2 = guard2
+            .find_or_find_insert_slot(h2, eq2, hash_entry)
+            .unwrap_or_else(|slot| unsafe {
+                // SAFETY: See first insert_in_slot call
+                guard2.insert_in_slot(h2, slot, (key2.clone(), SharedValue::new(insert_with())))
+            });
+
         (
             RefMut::Simple {
                 _guard: guard1,
-                key: key1,
-                value: value1,
+                bucket: bucket1,
             },
             RefMut::Simple {
                 _guard: guard2,
-                key: key2,
-                value: value2,
+                bucket: bucket2,
             },
         )
     }
```

Comment:
It's a very small microbenchmark, but this test does run noticeably faster after this PR (unscientific, noisy macbook):

```
hyperfine target/debug/deps/turbo_tasks_backend-c8ee3823c4b5c9c6 target/debug/deps/turbo_tasks_backend-4c1dc9d62e03b1cc
Benchmark 1: target/debug/deps/turbo_tasks_backend-c8ee3823c4b5c9c6
  Time (mean ± σ):      3.124 s ±  0.054 s    [User: 5.625 s, System: 7.997 s]
  Range (min … max):    3.068 s …  3.221 s    10 runs
 
Benchmark 2: target/debug/deps/turbo_tasks_backend-4c1dc9d62e03b1cc
  Time (mean ± σ):      2.286 s ±  0.050 s    [User: 3.858 s, System: 5.972 s]
  Range (min … max):    2.223 s …  2.367 s    10 runs
 
Summary
  target/debug/deps/turbo_tasks_backend-4c1dc9d62e03b1cc ran
    1.37 ± 0.04 times faster than target/debug/deps/turbo_tasks_backend-c8ee3823c4b5c9c6
```

Reply:
The new backend hits dash map even harder since it need to lookup for every task access... Could be a extra boost for it?